mease_lab_to_nwb/convert_syntalos/syntalosrecordinginterface.py

```diff
@@ -1,31 +1,67 @@
 """Authors: Cody Baker and Ben Dichter."""
 import numpy as np
+from tempfile import TemporaryFile
 
 from pynwb import NWBFile, TimeSeries
-from nwb_conversion_tools import IntanRecordingInterface
+from nwb_conversion_tools.baserecordingextractorinterface import BaseRecordingExtractorInterface
 from hdmf.data_utils import DataChunkIterator
 from hdmf.backends.hdf5.h5_utils import H5DataIO
 
+from .syntalosrecordingextractor import SyntalosRecordingExtractor
+
 
 def all_equal(lst: list):
     """Determine if all elements of a list are equal."""
     return len(set(lst)) == 1
 
 
-def get_block_info(shape, size):
-    """Calculate block information in binary file."""
-    assert len(shape) == 2, "The shape of the blocks in the binary file must have two dimensions!"
-    block_size = shape[1]
-    block_stop = size // block_size + 1
-    sl0 = block_size
-    sl1 = sl0 + size
-    return block_stop, sl0, sl1
+def write_accelerometer_data(nwbfile: NWBFile, recording, stub_test: bool = False):
+    """Add accelerometer data from a single rhd file to the NWBFile."""
+    accel_channels = np.array([ch for ch in recording._recordings[0]._recording._anas_chan if 'AUX' in ch['name']])
+    channel_conversion = [x['gain'] for x in accel_channels]
+    accel_channel_units = [x['units'] for x in accel_channels]
+    accel_channel_sampling_rate = [x['sampling_rate'] for x in accel_channels]
+
+    if not all_equal(channel_conversion):
+        raise NotImplementedError("Writing TimeSeries in NWBFiles with channel conversion factors is not supported!")
+    if not all([x['offset'] == 0 for x in accel_channels]):
+        raise NotImplementedError("Unable to support non-zero offsets for auxiliary channel data.")
+    if not all_equal(accel_channel_units):
+        raise NotImplementedError("Unequal auxiliary channel unit types are not yet supported.")
+    if not all_equal(accel_channel_sampling_rate):
+        raise NotImplementedError("Unequal auxiliary channel sampling rates are not yet supported.")
+    if not isinstance(recording._recordings[0]._recording._raw_data[accel_channels[0]['name']], np.memmap):
+        raise NotImplementedError("Writing accelerometer data for non-memory maps is not supported!")
+
+    conversion = channel_conversion[0]
+    all_memmaps = [
+        [x._recording._raw_data[accel_channels[j]['name']].flatten() for j in range(accel_channels.size)]
+        for x in recording._recordings
+    ]
+
+    all_accel_data = np.concatenate(np.moveaxis(np.array(all_memmaps), 2, 1))
+    if stub_test:
+        all_accel_data = all_accel_data[1:100, :]
```

Comment:
OK I know this is really minor, but should it be this?
```suggestion
        all_accel_data = all_accel_data[:100, :]
```